cli/src/cmd/forge/test.rs

```diff
@@ -160,100 +160,8 @@ impl TestArgs {
-        let mut runner = MultiContractRunnerBuilder::default()
-            .fuzzer(fuzzer)
-            .initial_balance(evm_opts.initial_balance)
-            .evm_spec(evm_spec)
-            .sender(evm_opts.sender)
-            .with_fork(utils::get_fork(&evm_opts, &config.rpc_storage_caching))
-            .build(project.paths.root, output, evm_opts)?;
-
-        if self.debug.is_some() {
-            self.filter.test_pattern = self.debug;
-            match runner.count_filtered_tests(&self.filter) {
-                1 => {
-                    // Run the test
-                    let results = runner.test(&self.filter, None)?;
-
-                    // Get the result of the single test
-                    let (id, sig, test_kind, counterexample) = results.iter().map(|(id, SuiteResult{ test_results, .. })| {
-                        let (sig, result) = test_results.iter().next().unwrap();
-
-                        (id.clone(), sig.clone(), result.kind.clone(), result.counterexample.clone())
-                    }).next().unwrap();
-
-                    // Build debugger args if this is a fuzz test
-                    let sig = match test_kind {
-                        TestKind::Fuzz(cases) => {
-                            if let Some(counterexample) = counterexample {
-                                counterexample.calldata.to_string()
-                            } else {
-                                cases.cases().first().expect("no fuzz cases run").calldata.to_string()
-                            }
-                        },
-                        _ => sig,
-                    };
-
-                    // Run the debugger
-                    let debugger = RunArgs {
-                        path: PathBuf::from(runner.source_paths.get(&id).unwrap()),
-                        target_contract: Some(utils::get_contract_name(&id).to_string()),
-                        sig,
-                        args: Vec::new(),
-                        debug: true,
-                        opts: self.opts,
-                        evm_opts: self.evm_opts,
-                    };
-                    debugger.run()?;
-
-                    Ok(TestOutcome::new(results, self.allow_failure))
-                }
-                n =>
-                    Err(
-                    eyre::eyre!("{} tests matched your criteria, but exactly 1 test must match in order to run the debugger.\n
-                        \n
-                        Use --match-contract and --match-path to further limit the search.", n))
-            }
-        } else {
-            let TestArgs { filter, .. } = self;
-            test(
-                runner,
-                verbosity,
-                filter,
-                self.json,
-                self.allow_failure,
-                (self.gas_report, config.gas_reports),
-            )
-        }
+    fn run(self) -> eyre::Result<Self::Output> {
+        custom_run(self, true)
```

Comment:
maybe we add a flag like --ignore-fuzz-tests so ppl can set this in their tests if they like?

Reply:
i think maybe save for a follow up pr

Reply:
yeah followup and the whole test run would benefit from another pass.